test/v1alpha1/service.go

```diff
@@ -112,22 +142,46 @@ func CreateRunLatestServiceReady(t *testing.T, clients *test.Clients, names *tes
 	}
 
 	t.Logf("Waiting for Service %q to transition to Ready.", names.Service)
-	if err := WaitForServiceState(clients.ServingAlphaClient, names.Service, IsServiceReady, "ServiceIsReady"); err != nil {
-		return nil, err
+	if err = WaitForServiceState(clients.ServingAlphaClient, names.Service, IsServiceReady, "ServiceIsReady"); err != nil {
+		return nil, nil, err
 	}
 
 	t.Log("Checking to ensure Service Status is populated for Ready service", names.Service)
 	err = validateCreatedServiceStatus(clients, names)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	var httpsTransportOption *spoof.TransportOption
+	if https {
+		tlsOptions := &v1alpha3.TLSOptions{
+			Mode:              v1alpha3.TLSModeSimple,
+			PrivateKey:        "/etc/istio/ingressgateway-certs/tls.key",
+			ServerCertificate: "/etc/istio/ingressgateway-certs/tls.crt",
+		}
+		servers := []v1alpha3.Server{{
+			Hosts: []string{"*"},
+			Port: v1alpha3.Port{
+				Name:     "standard-https",
+				Number:   443,
+				Protocol: v1alpha3.ProtocolHTTPS,
+			},
+			TLS: tlsOptions,
+		}}
+		domainName := strings.SplitN(names.URL.Host, ".", 2)[1]
```

Comment:
Do we still need to do the `splitN` here?
From what I saw, we get the suffix of `names.URL.Host`, and then we combine it with the service name, which is essentially the same as `names.URL.Host`. So I think we can use `names.URL.Host` directly here.

Reply:
Yeah I tried using `names.URL.Host` but it didn't work because it has the service name. domain name is `serving-tests.169.48.185.45.xip.io` but names.URL.Host is `hello-world-gzzyzjel.serving-tests.169.48.185.45.xip.io`

Reply:
what I mean here is we don't need to do `SplitN` and then combine the domain with service name in https://github.com/knative/serving/pull/5157/files/72501eed94be585c3d9060af4eabf2183b480185#diff-cd1624ee494bbca6fd2e1e1a908d0dd3R485. Instead, we can directly pass names.URL.Host to `setUpHTTPS` function.

Reply:
Ah got it. That makes sense. I've just pushed up the simplification.